src/openstudio_lib/OSDocument.cpp

```diff
@@ -1349,14 +1349,89 @@ void OSDocument::addStandardMeasures() {
   enable();
 }
 
-boost::optional<BCLMeasure> OSDocument::standardReportMeasure() {
-  std::string uid("a25386cd-60e4-46bc-8b11-c755f379d916");
-  boost::optional<BCLMeasure> result = LocalBCL::instance().getMeasure(uid);
-  if (result) {
-    return result;
+boost::optional<BCLComponent> OSDocument::getLocalComponent(const std::string& uid) const {
+  boost::optional<BCLComponent> result;
+  if (m_haveLocalBCL) {
+    try {
+      result = LocalBCL::instance().getComponent(uid);
+    } catch (const std::exception& e) {
+      LOG(Error, "Cannot access local BCL: " << e.what());
+      m_haveLocalBCL = false;
+    }
+  }
+  return result;
+}
+
+boost::optional<BCLComponent> OSDocument::getLocalComponent(const std::string& uid, const std::string& versionId) const {
+  boost::optional<BCLComponent> result;
+  if (m_haveLocalBCL) {
+    try {
+      result = LocalBCL::instance().getComponent(uid, versionId);
+    } catch (const std::exception& e) {
+      LOG(Error, "Cannot access local BCL: " << e.what());
+      m_haveLocalBCL = false;
+    }
```

Comment:
My only comment is whether we shouldn't just try to get the standardMeasure first and set m_haveLocalBCL once and for all to avoid having to do this check over and over in each method. That's not a huge deal though.

In fact, aren't we getting it in most cases already, unless we're just reloading? https://github.com/openstudiocoalition/OpenStudioApplication/blob/develop/src/openstudio_lib/OSDocument.cpp#L233-L235

Reply:
I guess it is pretty unlikely that `m_haveLocalBCL` would be true and then false in a single session, there is some overhead to the try catch but it's probably minimal.  I would rather just protect against any path in case we change the initialization order at some point.  Let's just merge as is, thanks for looking!